akka-actor-typed/src/main/scala/akka/actor/typed/receptionist/Receptionist.scala

```diff
@@ -7,12 +7,20 @@ package akka.actor.typed.receptionist
 import akka.actor.typed.{ ActorRef, ActorSystem, Extension, ExtensionId }
 import akka.actor.typed.internal.receptionist._
 import akka.annotation.DoNotInherit
-
 import scala.collection.JavaConverters._
 import scala.concurrent.duration._
 import scala.reflect.ClassTag
 
-class Receptionist(system: ActorSystem[_]) extends Extension {
+import akka.annotation.InternalApi
+
+abstract class Receptionist extends Extension {
```

Comment:
We can define such interfaces for all new typed extensions.

The reason for using abstract class instead of trait is the usual "support Java with Scala 2.11", since trait+object is missing static forwarder

Untyped extensions will not be supported. Anything doing `system.toUntyped` is not supported.